markdown/frameworks/src/javalin/JavalinRandomNumber.java

```diff
@@ -0,0 +1,66 @@
+package javalin;
+
+import io.javalin.Javalin;
+import io.javalin.http.Context;
+import io.javalin.http.Handler;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Random;
+
+public class JavalinRandomNumber {
+    public static void main(String[] args) {
```

Comment:
d.h. ich kann die sachen nur in genau der reihenfolge im browser aufrufen? oder registrierst du hier nur handler in der `app`, die in einem parallelen thread läuft? 

kontext wäre hilfreich :)  sonst muss ich mir das framework auch nochmal ansehen.

Reply:
Du registrierst den Kram. Kannst also Aufrufen wie du willst. 
An dieser stelle nimmt dir das Framework den ganzen typischen Web Krams ab.
Bin kein Webserver-Experte, aber da wird wohl irgendein Listener im Hintergrund aktiviert, der auf Anfragen reagiert und die dann den entsprechenden Handler übergibt. Ob das jetzt ein neuer Thread ist.... dont know. Muss ich mal nachschauen.


Ich pack dir nochmal ein paar Kommentare ran.

Reply:
ah, ok. die alternative erklärung wäre gewesen, dass die aufrufe in der reihenfolge wie im `main` kommen müssen. (aber gut, das wäre eher unwahrscheinlich, weil kaum hilfreich.)

Reply:
Hab ein paar Kommentare hinzugefügt.

Nochmal etwas ausführlicher:

Der Code erstellt einen Webserver der auf HTTP-Anfragen reagiert.

Mit `Javalin#get(String Path, Handler handler)` registrierst du einen Handler für eingehende HTTP-GET Anfragen auf dem angegeben  Pfad. 

Theoretisch kannst du auch Handler für PUT, POST, DELETE und was HTTP halt noch alles so kann erstellen- Hab ich jetzt mal nicht gemacht. Ohne Web know how ist das alles etwas Zuviel, dafür das es eigentlich nicht Thema ist. 

Wenn du einen Pfad/Link aufrufst, für den es keinen Handler gibt, dann wird der typische 404 returned.
 
Interessant ist jetzt noch das `Context` Objekt. Leider fehlt natürlich ausgerechnet dafür die JavaDoc. Alles was es gibt, ist das hier : https://javalin.io/documentation#context

Zusammengefasst lässt sich über das `Context` Objekt sagen: Es ist das große HTTP-Request Mopped und hat die ganzen Infos über die Anfrage (wer,wo,was,wann,warum, http protokolle, http boddy etc.) und ist auch das Tool um eine Antwort zurück zu geben `Context#result` .


Mit `Context#html(String html)` kann man auch HTML Code zurückgeben, der dann im Browser entsprechend gerendert wird. Ich glaube, das baue ich noch in das Beispiel ein, dann ist es irgendwie Runder.

Reply:
nice.

Reply:
Habe das Beispiel um HTML Krams erweitert.
Von der Logik ändert sich eigentlich nichts. 

Ich habe auch zwei Tasks zum starten im gradle hinzugefügt `helloWorld` und `randomNumber`


Info: 
Ich hatte jetzt ein paar mal das Problem, dass ich den Server nicht mit Crtl+c über die Shell beenden konnte. Da der Server den Port belegt und man so keinen neuen/anderen Server starten kann, habe ich in beiden Beispielen ein `localhost:8080/exit` Pfad eingebaut. Der beendet dann den Server.